src/main/java/net/apnic/rdap/rdap/RDAPError.java

```diff
@@ -0,0 +1,98 @@
+package net.apnic.rdap.rdap;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Class represent an error response in the RDAP protocol.
+ *
+ * @see https://tools.ietf.org/html/rfc7483
+ */
+public class RDAPError
+    extends RDAPObject
+{
+    private List<String> description = null;
```

Comment:
See I still disagree that this is the better way to go. We are trying to force immutability into a problem where it may not always be possible.

As an example if I make these structures immutable it means that there now has to be at least one but possibly more combinations of constructors because most if not all properties on some RDAPObjects are optional. So this means either passing null for the args in the constructor or taking different combinations.

On top of that if later on down the track we want to expand this and add more POJO's to encompass more of the possible objects available the factory would now need to implement another custom method just for that type. Or adding another member to an existing object would either require updating the constructor or adding a new constructor which has a knock on effect for all the using code.

Seems to me that adding immutability is trying to solve a problem that doesn't exist and is going to add more code to sections that just need an object. On top of that it starts to add coupling.

As above yes the use of generics has helped force the design into a gradual build approach of objects but I don't think that was the wrong approach. There are multiple parts that need to work on these objects and now they only have to concern themselves with what they care about. Without gradual setters they must take context from other places.

To summarise the RDAP spec specifically is designed as a tree structure for responses. This structure does not lend well to a build it all at once approach as you are going to always have different parts of the code with different concerns.